code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -98,11 +124,295 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == "zombie" || name == "skeleton")
```

Comment:
If I'm reading this correctly, initial checks what the name was on compile 

> This returns the original compile-time value of a variable. It could be used to reset a variable to its default value or to check if a variable has changed.

so with leatherman having been defined as a subtype of zombie, it'll come up as leatherman [numbers], which is kinda what I wanted to avoid